src/identity/components/MarketoAccountUpgradeOverlay.tsx

```diff
@@ -0,0 +1,227 @@
+  color: 'white',
+  textDecoration: 'underline',
+}
+
+// Types
+declare let window: WindowObjectPlusMarketoScript
+
+interface WindowObjectPlusMarketoScript extends Window {
+  MktoForms2?: MarketoScript
+}
+
+interface MarketoScript {
+  loadForm?: Function
+  getForm?: Function
+  whenReady?: Function
+}
+
+// Constants
+const BACKUP_CONTACT_SALES_LINK = 'https://www.influxdata.com/contact-sales-b/'
+const MARKETO_SERVER_INSTANCE = '//get.influxdata.com'
+const MARKETO_SUBSCRIPTION_ID = '972-GDU-533'
+const MARKETO_FORM_ID = 2826
+
+// Error Messaging
+enum MarketoError {
+  FormLoadingError = 'failed to load marketo account upgrade form',
+  FormSubmitError = 'failed to submit marketo account upgrade form',
+  ScriptFetchError = 'failed to fetch marketo account upgrade script',
+  ScriptRuntimeError = 'failed to run marketo account upgrade script',
+}
+
+// If marketo isn't working, still need the user to have some means of contacting sales.
+const SalesFormLink = (): JSX.Element => {
+  return (
+    <a
+      data-testid="use-sales-form--link"
+      href={BACKUP_CONTACT_SALES_LINK}
+      target="_blank"
+      style={popupLinkStyle}
+    >
+      Click here to contact sales.
+    </a>
+  )
+}
+
+export const MarketoAccountUpgradeOverlay: FC = () => {
+  const {onClose} = useContext(OverlayContext)
+  const dispatch = useDispatch()
+
+  const user = useSelector(selectUser)
+  const accountId = useSelector(selectCurrentAccountId)
+
+  const [scriptHasLoaded, setScriptHasLoaded] = useState(false)
+  const [formHasLoaded, setFormHasLoaded] = useState(false)
+
+  const handleCloseOverlay = () => {
+    // Deleting marketo object guards against inconsistent behavior if user closes then re-opens the overlay.
+    delete window.MktoForms2
+    onClose()
+  }
+
+  const handleError = useCallback(
+    (message: MarketoError, err: Error = new Error(message)) => {
+      if (message === MarketoError.FormSubmitError) {
+        dispatch(notify(marketoFormSubmitFailure(SalesFormLink)))
+      } else {
+        dispatch(notify(marketoLoadFailure(SalesFormLink)))
+      }
+
+      reportErrorThroughHoneyBadger(err, {
+        context: {
+          user,
+          accountId,
+        },
+        name: message,
+      })
+    },
+    [accountId, dispatch, user]
+  )
+
+  const handleFormSubmission = () => {
+    try {
+      window.MktoForms2.getForm(MARKETO_FORM_ID).submit()
+      dispatch(notify(marketoFormSubmitSuccess()))
+      onClose()
+    } catch (err) {
+      handleError(MarketoError.FormSubmitError, err)
+    }
+  }
+
+  const loadMarketoForm = useCallback(() => {
+    try {
+      if (!formHasLoaded) {
+        // API: https://developers.marketo.com/javascript-api/forms/api-reference/
+        window.MktoForms2?.loadForm(
+          MARKETO_SERVER_INSTANCE,
+          MARKETO_SUBSCRIPTION_ID,
+          MARKETO_FORM_ID,
+          () => {
+            // Marketo fields should remain hidden from user (CSS display: false)
```

Comment:
```ts
CSS display: false
```

shouldn't this be `display: none` ? Just want to make sure the actually CSS is doing `display: none;` since `display: false` doesn't real

Reply:
Yup, that comment should reflect what the scss file actually has and is actually doing, i.e., `display: none`. 👍